canon/src/id.rs

```diff
@@ -120,6 +120,17 @@ impl Id {
         T::decode(&mut source)
     }
 
+    /// Takes the bytes corresponding to this id out of the underlying store.
+    ///
+    /// If the Id is inlined, this is a no-op and returns Ok(None)
+    pub fn promote_bytes(&self) -> Result<Option<Vec<u8>>, CanonError> {
```

Comment:
`promote_bytes` is IMVHO a really confusing name. Even with the comment, I'm still not sure what exactly does, or more specifically why is called "promote". Could you explain to me a bit better? Do you have alternative candidate names for this method?

Reply:
Agreed that it needs better comment.

I chose promote since you're actually deleting the cached encoded bytes from the local store, and putting them instead in the persistant backend. So `delete` is not really accurate, but also not `get`.

Come to think of it `take_bytes` would be much better. I'll rename.